lib/Models/Catalog/Ows/OwsInterfaces.ts

```diff
@@ -6,6 +6,85 @@ export interface OnlineResource {
   "xlink:href": string;
 }
 
+type RangeClosureType = "closed" | "open" | "open-closed" | "closed-open";
+
+/** A range of values of a numeric parameter. This range can be continuous or discrete, defined by a fixed spacing between adjacent valid values. If the MinimumValue or MaximumValue is not included, there is no value limit in that direction. Inclusion of the specified minimum and maximum values in the range shall be defined by the rangeClosure. */
+export interface RangeType {
+  /** Specifies which of the minimum and maximum values are included in the range. Note that plus and minus infinity are considered closed bounds. */
+  readonly rangeClosure?: RangeClosureType;
+
+  /** Maximum value of this numeric parameter. */
+  readonly MaximumValue?: string;
+  /** Minimum value of this numeric parameter. */
+  readonly MinimumValue?: string;
+  /** The regular distance or spacing between the allowed values in a range. */
+  readonly Spacing?: string;
+}
+
+interface AllowedValuesType {
+  readonly Range?: RangeType | RangeType[];
+  readonly Value?: string | string[];
+}
+
+export interface DomainType {
+  /** Name or identifier of this quantity. */
+  name: string;
+  /** List of all the valid values and/or ranges of values for this quantity. For numeric quantities, signed values should be ordered from negative infinity to positive infinity. */
+  readonly AllowedValues: AllowedValuesType;
+}
+
+export interface RequestMethodType extends OnlineResource {
+  /** Optional unordered list of valid domain constraints on non-parameter quantities that each apply to this request method for this operation. If one of these Constraint elements has the same "name" attribute as a Constraint element in the OperationsMetadata or Operation element, this Constraint element shall override the other one for this operation. The list of required and optional constraints for this request method for this operation shall be specified in the Implementation Specification for this service. */
+  readonly Constraint?: DomainType | DomainType[];
```

Comment:
The PR consistently uses `ElemType | ElemType[]` in various places so it must be deliberate, but to me this feels like it's working against the type checker since it will be hard to statically determine if there is a single member or an array of members in an attribute. This seems confirmed by the code performing `x = Array.isArray(y) ? y : [y]` in a number of places.

I did a quick search and it seems possible to define a non-empty array type like this:
```typescript
type NonEmptyArray<T> = [T, ...T[]];

interface X {
  readonly Things?: NonEmptyArray<string>;
}
```
which I'm guessing should be able to represent the expected values?

If the intention is to be able to represent no constraint by not having the member, one constraint with the non-array type, >1 constraint with the array type, and the empty array signaling something different, that should be noted in some comment.

Reply:
This is just how the XML conversion library works, as it doesn't have/use schema information. When only one element is available, it will convert it to a single element. When multiple elements are available, it will convert them to an array. So this way, we get proper typing for later usage. 
I don't like those Array.isArray checks, but I don't see any other way besides writing a custom post-processor, and I don't see enough benefits to complicate things

Reply:
Ouch. Well, knowing that, I agree that a custom post-processor isn't worth the effort.